### PR TITLE
Revert changes to test regexes

### LINKED ISSUE #25867: BLD/CI: wheel building is failing
I think some PRs yesterday broke pandas-wheels: https://travis-ci.org/MacPython/pandas-wheels/jobs/510798584, likely all the same issues: https://travis-ci.org/MacPython/pandas-wheels/builds/510798582

if someone can you do a quick PR to update (not easy to test unfort).

```
______________________ TestFactorize.test_complex_sorting ______________________
[gw0] linux -- Python 3.6.6 /venv/bin/python
self = <pandas.tests.test_algos.TestFactorize object at 0x7f7cbf173710>
    def test_complex_sorting(self):
        # gh 12666 - check no segfault
        x17 = np.array([complex(i) for i in range(17)], dtype=object)
    
        msg = (r"unorderable types: {0} [<>] {0}".format(r"complex\(\)")
               if _np_version_under1p14 else
               r"'[<>]' not supported between instances of {0} and {0}".format(
                   "'complex'")
               )
        with pytest.raises(TypeError, match=msg):
>           algos.factorize(x17[::-1], sort=True)
E           AssertionError: Pattern 'unorderable types: complex\(\) [<>] complex\(\)' not found in ''>' not supported between instances of 'complex' and 'complex''
/venv/lib/python3.6/site-packages/pandas/tests/test_algos.py:237: AssertionError
_________________________ TestSafeSort.test_unsortable _________________________
[gw0] linux -- Python 3.6.6 /venv/bin/python
self = <pandas.tests.test_sorting.TestSafeSort object at 0x7f7cbecbce48>
    def test_unsortable(self):
        # GH 13714
        arr = np.array([1, 2, datetime.now(), 0, 3], dtype=object)
        msg = (r"unorderable types: ({0} [<>] {1}|{1} [<>] {0})".format(
                   r"int\(\)", r"datetime\.datetime\(\)")  # noqa: E126
               if _np_version_under1p14 else
               (r"'[<>]' not supported between instances of "
                r"({0} and {1}|{1} and {0})").format(
                    "'int'", r"'datetime\.datetime'")
               )
        with pytest.raises(TypeError, match=msg):
>           safe_sort(arr)
E           AssertionError: Pattern 'unorderable types: (int\(\) [<>] datetime\.datetime\(\)|datetime\.datetime\(\) [<>] int\(\))' not found in ''>' not supported between instances of 'int' and 'datetime.datetime''
```

### PR DESCRIPTION
**Pull request by WillAyd**
- [X] maybe closes #25867

### CODE REVIEW

**Comment by WillAyd** [comment]
Reverted to state before #25752

File: pandas/tests/test_algos.py [hunk 1]
```diff
@@ -228,11 +228,9 @@ def test_complex_sorting(self):
         # gh 12666 - check no segfault
         x17 = np.array([complex(i) for i in range(17)], dtype=object)
 
-        msg = (r"unorderable types: {0} [<>] {0}".format(r"complex\(\)")
-               if _np_version_under1p14 else
```
Comment (h-vetinari) [inline pandas/tests/test_algos.py]: better to just replace `if _np_version_under1p14 else` with `"|"`

File: pandas/tests/test_sorting.py [hunk 2]
```diff
@@ -414,13 +413,10 @@ def test_mixed_integer_from_list(self):
     def test_unsortable(self):
         # GH 13714
         arr = np.array([1, 2, datetime.now(), 0, 3], dtype=object)
-        msg = (r"unorderable types: ({0} [<>] {1}|{1} [<>] {0})".format(
-                   r"int\(\)", r"datetime\.datetime\(\)")  # noqa: E126
-               if _np_version_under1p14 else
```
Comment (h-vetinari) [inline pandas/tests/test_sorting.py]: Same here.

**Review by h-vetinari — CHANGES_REQUESTED** [review]
I find it curious that this passes - I've had persistent errors with the previous regexes. Sorry for breaking things.

I should add that I find the perfect matching of error messages to be a somewhat futile excercise, but I'm also guilty of that because in trying to be as correct as possible, I introduced the switch `_np_version_under1p14` which I observed. The failure in #25867 shows that this was not the correct distinction between the two parts.

In any case, the regexes I wrote are a true superset of the ones they replaced, and I think this should be kept in case the errors I encountered return. For this reason, I'd like to just remove the switch and turn it into a regex-or-clause.

**Comment by jreback** [comment]
@h-vetinari yeah not really sure why this fails the wheel builder but passes here.

**Comment by h-vetinari** [comment]
@jreback 
I had an open review request on this PR. Can you please wait with merging in such cases until it has at least been discussed?

**Comment by jreback** [comment]
@h-vetinari this was a straight revert and causing the CI to fail for several days. You should be familiar with how I work by now. We fix things step by step incrementally with orthogonal changes. you are welcome to propose changes if you want.

**Pull request merged**